CONTRIBUTING.md

```diff
@@ -1,30 +1,151 @@
 
-Message of commit must contain specific phrases, like `<type>(<package>): <commit-message> #<issue-id>` eg.`Feat(web-react): React Modal implementation #DS-331`
+### Commit Message Format
+
+We have very precise rules over how our Git commit messages must be formatted.
+This format leads to **easier to read commit history**.
+
+Each commit message consists of a **header**, a **body**, and a **footer**.
+
+```
+<header>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The `header` is mandatory and must conform to the [Commit Message Header](#commit-header) format.
+
+The `body` is optional but recommended for all commits except for those of type "Docs".
+When the body is present it must be at least 20 characters long and must conform to the [Commit Message Body](#commit-body) format.
+
+The `footer` is optional. The [Commit Message Footer](#commit-footer) format describes what the footer is used for and the structure it must have.
+
+#### <a name="commit-header"></a>Commit Message Header
+
+```
+<type>(<scope>): <short summary>
+  │       │             │
+  │       │             └─⫸ Summary in present tense. Sentence case. No period at the end.
+  │       │
+  │       └─⫸ Commit Scope: design-tokens|icons|web|web-react|web-twig|exporter-css|exporter-svg|
+  │                          ci|repo
+  │
+  └─⫸ Commit Type: Feat|Fix|Perf|Revert|Docs|Style|Refactor|Test|Chore|Deps
+```
+
+The `<type>` and `<summary>` fields are mandatory, the `(<scope>)` field is optional.
+
+##### Type
+
+Must be one of the following:
+
+- **Chore**: Changes to our CI configuration files and scripts (examples: CircleCi, SauceLabs) or changes that affect build system
+- **Deps**: Changes to dependencies
+- **Docs**: Documentation only changes
+- **Feat**: A new feature
+- **Fix**: A bug fix
+- **Perf**: A code change that improves performance
+- **Refactor**: A code change that neither fixes a bug nor adds a feature
+- **Revert**: Reverting code changes
+- **Test**: Adding missing tests or correcting existing tests
+- **Style**: Styling of components of changes to code style
+
+##### Scope
+
+The scope should be the name of the npm package affected (as perceived by the person reading the changelog generated from commit messages).
+
+The following is the list of supported scopes:
+
+- `design-tokens`
+- `icons`
+- `web`
+- `web-react`
+- `web-twig`
+- `exporter-css`
+- `exporter-svg`
+- `ci`
+- `repo`
+
+There are currently a few exceptions to the "use package name" rule:
+
+- `ci`: used for changes that affects Continuous Integration process and builds
+
+- `repo`: used for repository wide changes
+
+- none/empty string: useful for `test` and `refactor` changes that are done across all packages (e.g. `Test: Add missing unit tests`) and for docs changes that are not related to a specific package (e.g. `Docs: Fix typo in tutorial`).
```

Comment:
Are such changes added to all change logs? Because the top-level CHANGELOG (could it go there?) is currently just a signpost.

Reply:
If you omit the scope, the message will be in all changelogs, I think. Root changelog is not generated and accessed by Lerna.